### DISCUSSION TITLE
Hardhat Node Failing with Rust Panic: "Must be present as this is not a pending block"

### DISCUSSION THREAD

**Discussion by samiejumade** [NomicFoundation/hardhat#5814] [Troubleshooting]
Hello Hardhat community,

I'm encountering a persistent error when trying to run a Hardhat node. Here are the details:

Error:
When running 'npx hardhat node', I get a Rust panic with the message:
"thread 'tokio-runtime-worker' panicked at /Users/runner/work/edr/edr/crates/edr_evm/src/blockchain/remote.rs:232:14: Must be present as this is not a pending block"

Environment:
- Node.js version: 18.12.1
- Hardhat version: 2.22.13
- Operating System: macOS

Steps taken:
1. Cleaned Hardhat cache
2. Reinstalled dependencies
3. Updated packages
4. Simplified hardhat.config.js

Current hardhat.config.js:
```javascript
require("@nomiclabs/hardhat-waffle");

module.exports = {
  solidity: {
    compilers: [
      {
        version: "0.8.19",
        settings: {
          optimizer: {
            enabled: true,
            runs: 1000,
          },
        },
      },
    ],
  },
  networks: {
    hardhat: {
      forking: {
        url: "https://rpc.ankr.com/eth",
      },
      chainId: 1337,
    },
  },
};

Could this error be caused by incompatibilities between the Hardhat version (2.22.13) and the Node.js version (18.12.1) I'm using?

Are there any known issues with running Hardhat nodes on macOS that could lead to this Rust panic?

Is the chainId: 1337 configuration in the hardhat network settings causing any conflicts?

Are there any specific environment variables or configurations needed when running Hardhat with Rust dependencies?

Has anyone successfully resolved a similar "Must be present as this is not a pending block" error? If so, what steps did you take?

Are there any alternative RPC endpoints or forking configurations you'd recommend to potentially bypass this issue?

Could this error be related to the Solidity compiler version (0.8.19) specified in the config? Should I try a different version?

Are there any logs or additional debugging steps I can take to gather more information about this Rust panic?

**Comment by kanej** [comment]
This might be related to https://github.com/NomicFoundation/hardhat/issues/5816

Are you forking Sepolia?

**Reply by samiejumade** [reply]
No I am not forking sepolia, I am froking uniswap v3 pool for that i am using hardhat in local.
And here #5816 no solution has been given, although problem is related to that. can you give me solution here.

**Comment by kanej** [comment]
The EDR fix has been released as part of Hardhat `v2.22.14`:

https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.22.14
